Base/QTGUI/qSlicerSaveDataDialog.cxx

```diff
@@ -868,6 +870,19 @@ bool qSlicerSaveDataDialogPrivate::saveNodes()
         // Make sure an error message is added if saving returns with error
         snode->GetUserMessages()->AddMessage(vtkCommand::ErrorEvent,
           (qSlicerSaveDataDialog::tr("Cannot write data file: %1.").arg(file.absoluteFilePath())).toStdString());
+
+        // Add warning messages if the file name or path is too long
+        if (file.fileName().length() > vtkMRMLStorageNode::GetRecommendedFileNameLength())
+        {
+          snode->GetUserMessages()->AddMessage(vtkCommand::WarningEvent,
+            (qSlicerSaveDataDialog::tr("File name may be too long: %1.").arg(file.fileName())).toStdString());
+        }
+        if (file.absoluteFilePath().length() > 200)
```

Comment:
```suggestion
        // Maximum file path is 260 characters on most Windows systems. Warn the user if the path
        // is long and therefore may fail to be saved or cause compatibility issues later.
        // The warning limit is set to somewhat lower than the 260 limit, to add some safety margin
        // (e.g., to avoid issues when the file is moved into another folder).
        if (file.absoluteFilePath().length() > 200)
```